src/docs/release/breaking-changes/rendereditable-stop-painting-invalid-selection.md

```diff
@@ -0,0 +1,104 @@
+---
+title: Change RenderEditable to stop painting the caret when its selection is invalid
+description: RenderEditable no longer paints the caret when its selection is invalid.
+---
+
+## Summary
+
+When a text field's selection is `TextSelection.collapsed(offset: -1)`, or 
+a different range that contains negative positions, `RenderEditable` no 
+longer paints the caret or calls`onCaretChanged`.
+
+## Context
+
+The caret indicates the insertion point within the current text in an 
+active input field. Typically, when a new character is entered, the 
+caret stays immediately after it. In Flutter, the caret position is 
+represented by a collapsed selection. An invalid selection is a selection
+that contains negative text positions, such as `(-1, -1)`. When the 
+selection is invalid (which, for example, can happen when you omit the 
+`selection` argument when creating a `TextEditingValue`), usually the 
```

Comment:
you meant setting the texteditingvalue when the textfield has the focus? I wonder if we should do something about that or warn developer about this. I feel this situation is undesirable

Reply:
Agreed. Unfortunately we're defaulting to `(-1, -1)` in a lot of places so it is tricky to fix all of them without breaking many apps.